src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -462,18 +456,21 @@ void AddNativeLibrariesFromAssemblies (ZipArchive apk, string supportedAbis)
 						continue;
 					var data = ressozip.GetResourceData ();
 					using (var ms = new MemoryStream (data)) {
-						using (var zip = new ZipArchive (ms, ZipArchiveMode.Read)) {
-							foreach (var e in zip.Entries.Where (x => abis.Any (a => x.FullName.Contains (a)))) {
-								if (e.IsDirectory ())
+						Console.WriteLine ($"AddNativeLibrariesFromAssemblies {ressozip.Name}");
+						using (var zip = ZipArchive.Open (ms)) {
+							foreach (var e in zip.Where (x => abis.Any (a => x.Name.Contains (a)))) {
+								if (e.IsDirectory)
 									continue;
-								var key = e.FullName.Replace ("native_library_imports", "lib");
-								if (apk.ContainsEntry (key)) {
+								var key = e.Name.Replace ("native_library_imports", "lib");
+								if (apk.Any(k => k.Name  == key)) {
```

Comment:
Do we know what the performance implications of this `.Any()` invocation are? I haven't looked at what the `GetEnumerator()`/etc. code does, but if it involves lots of P/Invoke transitions, we might want to change this to use a "local" `HashSet<string>` "cache".